sympy/physics/mechanics/functions.py

```diff
@@ -756,6 +756,10 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
         if len(speeds) != len(set(speeds)):
             raise ValueError('Duplicate generalized speeds found, all '
                              'generalized speeds should be unique.')
+        if len(coordinates) + len(speeds) != len(
+            set(coordinates).union(speeds)):
+            raise ValueError('Some generalized coordinates are also defined as '
+                             'generalized speeds.')
```

Comment:
Are we happy this this error message is informative enough? Or do we want to try and also tell the reader which is duplicated?